star/test_suite/starspots/src/run_star_extras.f90

```diff
@@ -0,0 +1,413 @@
+
+         ! the extras functions in this file will not be called
+         ! unless you set their function pointers as done below.
+         ! otherwise we use a null_ version which does nothing (except warn).
+
+
+         fspot = s% x_ctrl(1)
+         xspot = s% x_ctrl(2)
+
+
+         s% extras_startup => extras_startup
+         s% extras_start_step => extras_start_step
+         s% extras_check_model => extras_check_model
+         s% extras_finish_step => extras_finish_step
+         s% extras_after_evolve => extras_after_evolve
+         s% how_many_extra_history_columns => how_many_extra_history_columns
+         s% data_for_extra_history_columns => data_for_extra_history_columns
+         s% how_many_extra_profile_columns => how_many_extra_profile_columns
+         s% data_for_extra_profile_columns => data_for_extra_profile_columns  
+
+         s% how_many_extra_history_header_items => how_many_extra_history_header_items
+         s% data_for_extra_history_header_items => data_for_extra_history_header_items
+         s% how_many_extra_profile_header_items => how_many_extra_profile_header_items
+         s% data_for_extra_profile_header_items => data_for_extra_profile_header_items
+
+
+         if (s% use_other_surface_PT) then
+            s% other_surface_PT => starspot_tweak_PT
+            !s% other_surface_PT => starspot_tweak_PT_Joyce
+         end if
+
+
+         if (s% use_other_mlt_results) then
+           s% other_mlt_results => YREC_spots_other_mlt_results
+         end if
+        
+
+      end subroutine extras_controls
+
+
+      subroutine extras_startup(id, restart, ierr)
+         integer, intent(in) :: id
+         logical, intent(in) :: restart
+         integer, intent(out) :: ierr
+         type (star_info), pointer :: s
+         ierr = 0
+         call star_ptr(id, s, ierr)
+         if (ierr /= 0) return
+         call test_suite_startup(s, restart, ierr)
+      end subroutine extras_startup
+      
+
+      integer function extras_start_step(id)
+         integer, intent(in) :: id
+         integer :: ierr
+         real(dp) :: mu_ideal_gas, R2, R_gas_constant !! pi can be deleted
+         type (star_info), pointer :: s
+         real(dp) :: power_he_burn, power_c_burn, power_neutrinos, &
+         center_h1, center_he4, ocz_top_mass, ocz_bot_mass, &
+         ocz_top_radius, ocz_bot_radius!, mass_difference_prev !! no!!
+         integer :: nz, j, i, k, k_ocz_top, k_ocz_bot, n_conv_bdy
+!         include 'formats'
+         ierr = 0
+         call star_ptr(id, s, ierr)
+         if (ierr /= 0) return
+         extras_start_step = 0
+
+         !! to use constants as defined in MESA, import const_def and call
+         !! variables by their names there
+
+         ! set PB_i here and then other_mlt will know about it, because 
+         ! quantities set here are carried over the course of a timestep, 
+         ! AKA all newton iterations
+         
+         sb_sigma = boltz_sigma !5.67d-5 !!cgs units
```

Comment:
Why not just replace `sb_sigma` with `boltz_sigma` instead of defining another variable? (Same goes for `R_gas_constant = cgas` below.)